src/auth/service.py

```
@@ -37,93 +131,252 @@ async def google_oauth_flow(code: str, session: AsyncSession) -> OAuthCallbackRe
+    네이티브 SDK에서 받은 id_token을 검증하고 JWT 토큰 발급.
+    """
+    async with httpx.AsyncClient() as client:
+        response = await client.get(
+            constants.GOOGLE_TOKENINFO_URL,
+            params={"id_token": id_token},
         )
 
-    # New user - create active user with auto-accepted terms
-    new_user = await create_oauth_user(
+        if response.status_code != 200:
+            raise InvalidTokenError("Invalid Google ID token")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)